Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -793,17 +793,19 @@ extension PurchasesOrchestrator: StoreKit2TransactionListenerDelegate {
         _ listener: StoreKit2TransactionListener,
         updatedTransaction transaction: StoreTransactionType
     ) async throws {
-        await Async.call { completion in
-            self.finishTransactionIfNeeded(transaction) { @MainActor in
-                completion(())
-            }
-        }
-
         let isRestore = self.systemInfo.observerMode
 
+        precondition(!self.finishTransactions)
```

Comment:
This line I'm going to remove, I just want to see if any of the integration tests that aren't observer mode would be affected by this change.